### PR TITLE
FreeBSD build little updates.

### PR DESCRIPTION
**Pull request by devnexen**
Both clang/gcc toolchains are recognised under this system.
Little hotspot method implementation to match a bit Linux's.

### CODE REVIEW

**Review by battleblow — COMMENTED** [review]
Please target the bsd-port branch with any pull requests.  The master branch is a a pristine branch for pulling in the AdoptOpenJDK upstream.

File: src/hotspot/os/bsd/os_bsd.cpp [hunk 1]
```diff
@@ -1560,7 +1560,13 @@ void os::print_os_info(outputStream* st) {
 }
 
 void os::pd_print_cpu_info(outputStream* st, char* buf, size_t buflen) {
-  // Nothing to do for now.
+  char model[100];
```
Comment (battleblow) [inline src/hotspot/os/bsd/os_bsd.cpp]: You don't need to declare a buffer to store the model name, the caller passes one in.

File: src/hotspot/os/bsd/os_bsd.cpp [hunk 2]
```diff
@@ -1560,7 +1560,13 @@ void os::print_os_info(outputStream* st) {
...
+  size_t size = sizeof(model);
+  int mib[] = { CTL_HW, HW_MODEL };
```
Comment (battleblow) [inline src/hotspot/os/bsd/os_bsd.cpp:1564]: How portable is this across BSDs?

File: make/autoconf/toolchain.m4 [hunk 3]
```diff
@@ -42,6 +42,7 @@ VALID_TOOLCHAINS_solaris="solstudio"
 VALID_TOOLCHAINS_macosx="gcc clang"
 VALID_TOOLCHAINS_aix="xlc"
 VALID_TOOLCHAINS_windows="microsoft"
+VALID_TOOLCHAINS_bsd="gcc clang"
```
Comment (battleblow) [inline make/autoconf/toolchain.m4]: Please put this change and the CPU model change in a separate commit since they are unrelated to each other.

File: src/hotspot/os/bsd/os_bsd.cpp [hunk 4]
```diff
@@ -1560,7 +1560,13 @@ void os::print_os_info(outputStream* st) {
...
+  if (sysctl(mib, 2, model, &size, NULL, 0) == 0) {
+    st->print("model name: %s", model);
```
Comment (battleblow) [inline src/hotspot/os/bsd/os_bsd.cpp]: Looking at the Linux version, I think "CPU Model: %s" would be more consistent.

File: src/hotspot/os/bsd/os_bsd.cpp (hunk 2 above)
Comment (devnexen) [reply src/hotspot/os/bsd/os_bsd.cpp:1564]: It is portable included macOS and even DragonflyBSD.
